core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/AmazonAsyncHttpClient.java

```diff
@@ -169,6 +172,9 @@ public <OutputT> CompletableFuture<OutputT> execute(
                                 .then(ApplyUserAgentStage::new)
                                 .then(MergeCustomHeadersStage::new)
                                 .then(MergeCustomQueryParamsStage::new)
+                                .then(HttpChecksumRequiredStage::new)
+                                .then(AsyncRequestBodyHttpChecksumTrailerStage::new)
+                                .then(HttpChecksumInHeaderStage::new)
```

Comment:
Do we need three different stages for checksums or can we combine them into one stage?

Reply:
right, i wasn't sure since we had separate interceptors. aggregating into one stage